packages/toolpad-core/src/types.ts

```diff
@@ -75,7 +75,15 @@ export type BindableAttrEntries = [string, BindableAttrValue<any>][];
 export type SlotType = 'single' | 'multiple' | 'layout';
 
 export interface ValueTypeBase {
-  type: 'string' | 'boolean' | 'number' | 'object' | 'array' | 'element' | 'event';
+  type:
+    | 'string'
+    | 'boolean'
+    | 'number'
+    | 'object'
+    | 'array'
+    | 'element'
+    | 'elementIterator'
```

Comment:
It wouldn't be possible to come up with a more generic construct, that allows us to expand templating to other situations? I'm thinking custom column types. Or custom autocomplete items.

Reply:
Do you have more specific examples of limitation this approach has that wouldn't work for those use cases?

**Custom column types:** if this would be a Datagrid feature, it might deserve its own specific approach?
**Custom autocomplete items:** what this new `elementIterator` does is allow us to render a template n number of times based on the layout of the first item, and wrap those items with whatever we want to (in the case of this `List`, a `ListItem`). So it should be possible to use an `elementIterator` for custom autocomplete items?

Reply:
There are two concepts here:

* The concept of "a template", which describes what this prop holds. a template that can be used to render multiple elements, based ona  common template
* The concept of "a template editing UI". This is what the "slot" control does for the `element` type. It defines how a user interacts with the property.

It seems to me that you are bundling these concepts together here in the `elementIterator` type. My expectation would be that there is a `template` type. and something like a `dragAndDrop` control. That way we separate the logic of what the prop holds from how it's being manipulated. e.g. you can have a single `number` type, that can be manipulated by several different controls.

Reply:
But for example the `Paper` component is already also using a `layoutSlot` for the same layout building experience, even though it does not iterate, so they're already separate concepts? Maybe it's just a question of naming things differently?
We can clear up these details later synchronously though and finish the discussion about this.